### PR TITLE
Return scalar multiplication in outer product if rank of one tenser i…

### PR DESCRIPTION
**Pull request by jziub**
This intends to fix [#196 ](https://github.com/google/TensorNetwork/issues/196)

### CODE REVIEW

**Comment by jziub** [comment]
@googlebot I signed it!

**Comment by jziub** [comment]
@googlebot I fixed it.

File: tensornetwork/backends/base_backend.py [hunk 1]
```diff
@@ -324,3 +324,15 @@ def eigsh_lanczos(
...
+    """
+      Return the default multiplication of `tensor`.
+      A backend can override such implementation.
+      Args:
+        tensor1: A tensor.
+        tensor2: A tensor.
+      Returns:
+        Tensor
+    """
+    return tensor1 * tensor2
```
Comment (MichaelMarien) [inline tensornetwork/backends/base_backend.py]: I'd prefer to raise an NotImplementedError here instead and add it to all the separate backends even if the implementation is identical. Anyway, definitely we need a test for this wherever it ends up.

File: tensornetwork/tests/network_test.py [hunk 2]
```diff
@@ -230,6 +230,16 @@ def test_outer_product(backend):
   assert d.name == "D"
 
 
+def test_outer_product_without_legs(backend):
+  a = tn.Node(np.ones((2, 4, 5)), name="A", backend=backend)
+  b = tn.Node(np.ones(()), name="B", backend=backend)
+  c = tn.outer_product(a, b, name="C")
+
```
Comment (MichaelMarien) [inline tensornetwork/tests/network_test.py]: perhaps test the other order also the arguments of outer product?

File: tensornetwork/backends/base_backend.py (hunk 1 above)
Comment (Thenerdstation) [reply tensornetwork/backends/base_backend.py]: +1

File: tensornetwork/tests/network_test.py (hunk 2 above)
Comment (Thenerdstation) [reply tensornetwork/tests/network_test.py]: +1

**Comment by jziub** [comment]
Closed this pull request as the commits were overwritten. Will send in a new pull request regarding this. Sorry about the confusion!

**Pull request closed without merging**